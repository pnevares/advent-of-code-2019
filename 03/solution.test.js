const solution = require("./solution");

describe("solution 01", () => {
  describe("can draw a single line", () => {
    it("should handle paths that go right", () => {
      expect(solution([["R4"]])).toMatchSnapshot();
    });
    it("should handle paths that go left", () => {
      expect(solution([["L4"]])).toMatchSnapshot();
    });
    it("should handle paths that go up", () => {
      expect(solution([["U4"]])).toMatchSnapshot();
    });
    it("should handle paths that go down", () => {
      expect(solution([["D4"]])).toMatchSnapshot();
    });
  });
  describe("can draw two lines", () => {
    it("should handle [D4, R4]", () => {
      expect(solution([["D4", "R4"]])).toMatchSnapshot();
    });
  });
});
