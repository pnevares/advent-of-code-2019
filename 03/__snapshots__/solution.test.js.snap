// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`solution 01 can draw a single line should handle paths that go down 1`] = `
"
o
|
|
|
|
"
`;

exports[`solution 01 can draw a single line should handle paths that go left 1`] = `
"
----o
"
`;

exports[`solution 01 can draw a single line should handle paths that go right 1`] = `
"
o----
"
`;

exports[`solution 01 can draw a single line should handle paths that go up 1`] = `
"
|
|
|
|
o
"
`;

exports[`solution 01 can draw two lines should handle [D4, R4] 1`] = `
"
o....
|....
|....
|....
|----
"
`;

exports[`solution 01 can draw two lines should handle [R4, D4] 1`] = `
"
o----
....|
....|
....|
....|
"
`;

exports[`solution 01 can draw two lines should handle [R4, U4] 1`] = `
"
....|
....|
....|
....|
o----
"
`;

exports[`solution 01 can draw two lines should handle [U4, L4] 1`] = `
"
----|
....|
....|
....|
....o
"
`;
